go/vt/vttablet/tabletserver/dt_executor.go

```diff
@@ -80,12 +84,48 @@ func (dte *DTExecutor) Prepare(transactionID int64, dtid string) error {
 		return vterrors.VT10002("cannot prepare the transaction on a reserved connection")
 	}
 
+	// Fail Prepare if any query rule disallows it.
+	// This could be due to ongoing cutover happening in vreplication workflow
```

Comment:
Does vtgate attempt to buffer these queries?

Reply:
This is a prepare call for a distributed transaction that is trying to be committed. If this call fails, then the transaction will be rollbacked.